components/builder-depot/src/backend/s3.rs

```diff
@@ -103,6 +102,24 @@ impl S3Handler {
         }
     }
 
+    // This function checks whether an uploaded file
+    // exists in the configured s3 bucket. It should
+    // only get called from within an upload future.
+    fn object_exists(&self, object_key: &str) -> Result<()> {
+        let mut request = HeadObjectRequest::default();
+        request.bucket = self.bucket.clone();
+        request.key = object_key.to_string();
+
+        match self.client.head_object(&request).sync() {
+            Ok(object) => {
+                info!("Verified {} was written to minio!", object_key);
+                debug!("Head Object check returned: {:?}", object);
+                Ok(())
+            }
+            Err(e) => Err(Error::HeadObject(e)),
```

Comment:
If object verification fails, we don't seem to be logging anything anywhere or failing the upload later on..?

Reply:
ah ok, looks like the and_then result will get bubbled up.. probably would be good to log a warning message here as well